ui/de-lib/src/main/java/org/iplantc/de/client/models/sharing/Sharing.java

```diff
@@ -9,29 +14,67 @@
  */
 public class Sharing {
```

Comment:
This class is already used to serialize json data. In the `SharingPresenter` callbacks. Along with my other comments above, serializing/deserializing of this class can be performed as-is in the service layer. When that happens, I think the (de)serializing logic can be simplified with the use of turning this class into an overlay type.